### ISSUE TITLE
HueSmartButton

### ISSUE THREAD

**Issue by oscfor** [xaviml/controllerx#498]
# New Device Support

## Device Information

- Device Model: [ ROM001 ]
- Device Description: [ HueSmartButton]
- Device Manufacturer: [ Hue ]

#### Actions

- ``: Sent when button xyz is pressed
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"press","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'press'
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"release","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:19MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'release'
Info 2022-05-31 20:22:20MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"on","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:20MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:20MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'on'

- `button_xyz_hold`: Sent when button xyz is held
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"press","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'press'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":87,"step_size":30,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"hold","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:35MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'hold'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":87,"step_size":56,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"hold","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:36MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'hold'
Info 2022-05-31 20:22:37MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"release","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:37MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-05-31 20:22:37MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'release'

**Comment by xaviml** [comment]
Hi @oscfor ,

I am answering your comment from https://github.com/xaviml/controllerx/issues/284#issuecomment-1142507662:

Could you please let me know which ControllerX was working and now is not?

Also, you please place your configuration here with proper indentation:

\`\`\`yaml
Your configuration here
\`\`\`

You can do the same for the logs:
\`\`\`
Your logs here
\`\`\`

Please add AppDaemon logs as well to show is not working.

Regards,
Xavi M.

**Comment by oscfor** [comment]
Hi @xaviml ,

Sorry, I'm not that used to github. 

ControllerX was working at version 4.18.2

```yaml
matsalsdimmer:
  module: controllerx
  class: HueSmartButtonLightController
  controller: zigbee2mqtt/matsalsdimmer/action
  integration: mqtt
  light: light.matsal_taklampa
```
**Button press**
```
 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"press","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'press'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"release","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'release'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"off","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:23:31MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'off'
```
**Button hold**
```
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"press","battery":100,"linkquality":94,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":94,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'press'
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":91,"step_size":30,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:11MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"hold","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'hold'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":91,"step_size":56,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:12MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"hold","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":83,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'hold'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":87,"step_size":56,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"hold","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":91,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:13MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'hold'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"skip_backward","battery":100,"linkquality":87,"step_size":56,"transition_time":9,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'skip_backward'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"release","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer', payload '{"action":"","battery":100,"linkquality":87,"update":{"state":"idle"},"update_available":null}'
Info 2022-06-02 21:24:14MQTT publish: topic 'zigbee2mqtt/matsalsdimmer/action', payload 'release'
```
AppDaemon
```
2022-06-02 21:25:42.109671 WARNING matsalsdimmer: Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/appdaemon/app_management.py", line 163, in initialize_app
    await init()
  File "/config/appdaemon/apps/controllerx/cx_core/controller.py", line 103, in initialize
    await self.init()
  File "/config/appdaemon/apps/controllerx/cx_core/type/light_controller.py", line 148, in init
    await super().init()
  File "/config/appdaemon/apps/controllerx/cx_core/type_controller.py", line 66, in init
    await super().init()
  File "/config/appdaemon/apps/controllerx/cx_core/release_hold_controller.py", line 27, in init
    await super().init()
  File "/config/appdaemon/apps/controllerx/cx_core/controller.py", line 118, in init
    default_actions_mapping = self.get_default_actions_mapping(self.integration)
  File "/config/appdaemon/apps/controllerx/cx_core/controller.py", line 236, in get_default_actions_mapping
    raise ValueError(f"This controller does not support {integration.name}.")
ValueError: This controller does not support mqtt.
2022-06-02 21:25:42.109835 WARNING matsalsdimmer: ------------------------------------------------------------
```

**Comment by xaviml** [comment]
Hi @oscfor,

Thank you for the logs and the further details. Did you use to have the same configuration? I can just say that `HueSmartButtonLightController` does not have Zigbee2MQTT integration (and mqtt integration defaults to zigbee2mqtt mappings). Not even in 4.18.2 had a mapping, so even if you go back to v4.18.2, it will not work with that configuration. You can see the device mapping [here](https://github.com/xaviml/controllerx/blob/v4.18.2/apps/controllerx/cx_devices/phillips.py#L107-L114) and you will see that there is only deCONZ mapping. You can even see in [this comment](https://github.com/xaviml/controllerx/issues/284#issuecomment-875765390) that the integration for Z2M was never added to this controller due to missing information.

Did you probably use to use deCONZ and migrate to Zigbee2MQTT? Or maybe another controller?

Regards,
Xavi M.

**Comment by oscfor** [comment]
Hi,

It was about two years ago I migrated from deconz. I am 100% sure I was able to use the HueButton, which I purchased in March this year. Never mind though. Is the information provided enough to get it added now?

Kind Regards

Oscar

**Comment by xaviml** [comment]
Hi @oscfor,

Yes, sorry, it is more than enough! I will also leave this link for reference: https://www.zigbee2mqtt.io/devices/8718699693985.html

I will add this for the next release. However, in the meantime, you should be able to use with `mapping`:

```yaml
matsalsdimmer:
  module: controllerx
  class: HueSmartButtonLightController
  controller: zigbee2mqtt/matsalsdimmer/action
  integration: mqtt
  light: light.matsal_taklampa
  mapping:
    "on": "on"  # Maybe it makes more sense to be "toggle"
    "off": "off"  # Maybe it makes more sense to be "toggle"
    skip_backward:  on_full_brightness
    skip_forward: on_min_brightness
    press: toggle
    hold: hold_brightness_toggle
    release: release
```

Also, if you could please leave in here what actions you perform to see the events above, I would appreciate since it would help me create the default mapping. For example:
- "on": When I click the button once...
- "off": When I click the button once...
- skip_backward: ....
- skip_forward: ....
- press: ....
- hold: When I hold the button
- release: When I release the button after being held


Regards,
Xavi M.

**Comment by xaviml** [comment]
Hi @oscfor,

I just released a beta version [v4.21.0b0](https://github.com/xaviml/controllerx/releases/tag/v4.21.0b0) with the support of Z2M for the Hue Smart Button.

You can download from HACS enabling beta versions on ControllerX. Please, let me know if you find any problems. I will be releasing the non-beta version next week or the next.

Regards,
Xavi M.

**Comment by oscfor** [comment]
On my initial test it seems to be working like a charm! Thanks a bunch!
